File: client/utils/adherence.js
// Shared thresholds for coloring an adherence percentage — a client and
// their coach must read the same percentage as the same color, and every
// adherence surface (diary history, the nutrition builder's plan badge, the
// Food Diary modal) reuses this instead of redefining its own. Adherence is
// capped at 100 server-side (foodDiaryStats.ts) -- eating past a goal is
// still fully adherent, not "over" -- so only the low end needs a tier.
// Mirrored in mobile's shared/utils/adherence.dart -- no shared package
// across web/mobile exists in this repo, so keep both in sync.
const ADHERENCE_GOOD_THRESHOLD = 85;
const ADHERENCE_WARNING_THRESHOLD = 60;

export function adherenceColor(pct) {
    if (pct === null || pct === undefined) return "text-muted-foreground";
    if (pct >= ADHERENCE_GOOD_THRESHOLD) return "text-emerald-500";
    if (pct >= ADHERENCE_WARNING_THRESHOLD) return "text-amber-500";
    return "text-destructive";
}

// Same tiers, expressed as a HeroUI Chip `color` value.
export function adherenceChipColor(pct) {
    if (pct === null || pct === undefined) return "default";
    if (pct >= ADHERENCE_GOOD_THRESHOLD) return "success";
    if (pct >= ADHERENCE_WARNING_THRESHOLD) return "warning";
    return "danger";
}

File: mobile/lib/shared/utils/adherence.dart
import 'package:flutter/material.dart';

import '../../core/theme/app_theme.dart';

/// Text color tier for an adherence percentage — every adherence surface
/// (diary checklist, diary history) reuses this so a percentage always
/// reads the same color. Adherence is capped at 100 server-side (a client
/// eating past a goal is still fully adherent, not "over"), so only the
/// low end needs a tier. Mirrors web's `utils/adherence.js` -- no shared
/// package across web/mobile exists in this repo, so keep both in sync.
const _adherenceGoodThreshold = 85;
const _adherenceWarningThreshold = 60;

Color adherenceColor(BuildContext context, int? pct) {
  if (pct == null) return context.appColors.mutedForeground;
  if (pct >= _adherenceGoodThreshold) return context.appColors.success;
  if (pct >= _adherenceWarningThreshold) return context.appColors.warning;
  return Theme.of(context).colorScheme.error;
}
